libsolidity/analysis/NameAndTypeResolver.h

```diff
@@ -44,6 +44,8 @@ class ErrorReporter;
 namespace solidity::frontend
 {
 
+using DeferredDeclarations = std::vector<std::tuple<std::string, langutil::SourceLocation, std::shared_ptr<DeclarationContainer const>>>;
```

Comment:
I did not look at lifetimes yet to check if regular pointers would be sufficient.

Reply:
the lifetimes are bound to NameAndTypeResolver's lifetime which is destroyed at the end of the compilation process.